### PR TITLE
Add changelog for 2.16.4

### PR DESCRIPTION
**Pull request by turbo**

### CODE REVIEW

File: docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.16.4.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+""
+
+*   Improved support for flow through captured variables that properly adheres to inter-procedural control flow.
+*   We no longer make use of CodeQL database stats, which may affect join-orders in custom queries. It is therefore recommended to test performance of custom queries after upgrading to this version.
+
+Golang
+""""""
+
+*   We have significantly improved the Go autobuilder to understand a greater range of project layouts, which allows Go source files to be analysed that could previously not be processed.
+*   Go 1.22 has been included in the range of supported Go versions.
```
Comment (adityasharad) [inline docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.16.4.rst:121]: When debugging earlier today with @angelapwen I wondered whether we should have a similar entry for C# (C# 12 and .NET 8 support are now documented). We do mention C# 12 in the detailed changelog items below.

Comment (turbo) [reply docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.16.4.rst:121]: Is C# 12 new with this release?

**Review by saritai — APPROVED** [review]
LGTM!

File: docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.16.4.rst (hunk 1 above)
Comment (turbo) [reply docs/codeql/codeql-overview/codeql-changelog/codeql-cli-2.16.4.rst:121]: I think we're good to proceed without. Let's establish an internal guidance for these items soon, then we can be more strict about it. I'll mention it in the blog post of course.

**Pull request merged**